src/Compilers/CSharp/Portable/Compiler/TypeCompilationState.cs

```diff
@@ -68,6 +70,12 @@ internal MethodWithBody(MethodSymbol method, BoundStatement body, ImportChain im
 
         public LambdaFrame staticLambdaFrame;
 
+        /// <summary>
+        /// A graph of method->method references for this(...) constructor initializers.
+        /// Used to detect and report initializer cycles.
+        /// </summary>
+        Dictionary<MethodSymbol, Tuple<MethodSymbol, CSharpSyntaxNode>> _constructorInitializers;
```

Comment:
A full `Dictionary<TKey, TValue>` seems wasteful here because the majority case won't have more than a few entries. Do we have a frugal dictionary we could use here instead?